include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -9,3 +9,119 @@
 // This describes the Calyx structures.
 //
 //===----------------------------------------------------------------------===//
+
+def ComponentOp : CalyxOp<"component", [
+    Symbol,
+    FunctionLike,
+    IsolatedFromAbove,
+    // TODO(Calyx): Need to eventually expose the output ports.
+    NoTerminator,
+    SingleBlock
+  ]> {
+  let summary = "Calyx Component";
+  let description = [{
+    The "calyx.component" operation represents an overall
+    Calyx component containing:
+    (1) In- and output port definitions
+        that define the interface.
+    (2) The cells, wires, and control schedule.
+    ```mlir
+      calyx.component @MyComponent(%in1: i32) -> (%out1: i8) {
+        calyx.cells { ... }
+        calyx.wires { ... }
+        calyx.control { ... }
+      }
+    ```
+  }];
+
+  // TODO(Calyx): Allow explicit port naming?
+  let arguments = (ins);
```

Comment:
I see the parser sets some port name attributes, would it make sense to specify those here as well? As far as I know that would just help ODS generate some builders that accept such attributes. The parser is building up an `OperationState` directly, but those builders could be useful for other MLIR tools that might create this op. Also, having them in ODS helps document the name and type of the attributes.

Reply:
Done.